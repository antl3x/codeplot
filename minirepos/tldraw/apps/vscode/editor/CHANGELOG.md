# @tldraw/vscode-editor

## 1.13.1-alpha.0

### Patch Changes

- Release day!

## 1.13.0

### Minor Changes

- - Adds missing Arabic translations for dialogs. @abedshamia
  - Updates core-example. @brydenfogelman
  - Updates Polish translations. @adan2013
  - Adds missing Aria-Labels. @KDSBrowne
  - Improves Japanese translation. @yashkumarbarot
  - Fixes height and width in app.viewport. @hiroshisuga
  - Improves labels on StlyeMenu @proke03
  - Adds missing tooltips to undo / redo buttons. @proke03

## 1.12.0

### Minor Changes

- - Improve middle mouse panning
  - Fix bug with assets in VS Code plugin
  - Improve performance of draw-style shapes
  - Fix bug with creating assets
  - Fix bug with text align in labels when outputting images
  - Fix bug with middle mouse panning on Linux
  - Fix bug with zoom shortcuts on number pad
  - Fix bug with draw and erase direction when holding shift

## 1.11.0

### Minor Changes

- d919bd27: Bump dependencies, add international support.

## 1.11.0-next.0

### Minor Changes

- Bump dependencies, add international support.

## 1.10.2

### Patch Changes

- Fix tldraw assets for vscode extension.

## 1.10.1

### Patch Changes

- Fix build.

## 1.10.0

### Minor Changes

- Fix build error in extension.

## 1.9.0

### Minor Changes

- Bump underlying packages.

## 1.8.0

### Minor Changes

- c09d6a3a: Adds text field for page rename, undo buttons on all screen sizes, arrow behavior with alt key.

## 1.7.1

### Patch Changes

- Fix bug with missing parents / children.
