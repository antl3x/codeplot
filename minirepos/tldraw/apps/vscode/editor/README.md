<div style="text-align: center; transform: scale(.5);">
  <img src="https://github.com/tldraw/tldraw/raw/main/assets/card-repo.png"/>
</div>

# @tldraw/vscode-editor

The app for the tldraw VS Code Extension.

See the README at `vscode` for more about this project.
