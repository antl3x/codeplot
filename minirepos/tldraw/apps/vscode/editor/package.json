{
	"name": "@tldraw/vscode-editor",
	"description": "An an editor for the tldraw vscode extension.",
	"version": "2.0.0-alpha.0",
	"private": true,
	"packageManager": "yarn@3.5.0",
	"author": {
		"name": "tldraw GB Ltd.",
		"email": "hello@tldraw.com"
	},
	"homepage": "https://tldraw.dev",
	"license": "Apache-2.0",
	"repository": {
		"type": "git",
		"url": "https://github.com/tldraw/tldraw"
	},
	"bugs": {
		"url": "https://github.com/tldraw/tldraw/issues"
	},
	"keywords": [
		"tldraw",
		"drawing",
		"app",
		"development",
		"whiteboard",
		"canvas",
		"infinite"
	],
	"scripts": {
		"build": "yarn run -T tsx scripts/build.ts",
		"dev": "yarn run -T tsx scripts/dev.ts",
		"clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist",
		"lint": "yarn run -T tsx ../../../scripts/lint.ts"
	},
	"devDependencies": {
		"@tldraw/assets": "workspace:*",
		"@tldraw/tldraw": "workspace:*",
		"@types/fs-extra": "^11.0.1",
		"@types/node": "^17.0.14",
		"@types/react": "^18.0.24",
		"@types/react-dom": "^18.0.6",
		"@types/react-router-dom": "^5.1.8",
		"concurrently": "7.0.0",
		"create-serve": "1.0.1",
		"dotenv": "^16.0.3",
		"esbuild": "^0.18.4",
		"fs-extra": "^11.1.0",
		"lazyrepo": "0.0.0-alpha.27",
		"react": "^18.2.0",
		"react-dom": "^18.2.0",
		"tslib": "^2.4.0"
	},
	"gitHead": "a7dac0f83ad998e205c2aab58182cb4ba4e099a6"
}
