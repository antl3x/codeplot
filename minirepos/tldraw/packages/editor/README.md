# tldraw/tldraw

## Benchmark

To run the benchmarks

```
yarn workspace @tldraw/tldraw benchmark
```

Or

```
yarn workspace @tldraw/tldraw benchmark "file_search_string"
```
