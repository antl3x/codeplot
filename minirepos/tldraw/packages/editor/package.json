{
	"name": "@tldraw/editor",
	"description": "A tiny little drawing app (editor).",
	"version": "2.0.0-alpha.19",
	"packageManager": "yarn@3.5.0",
	"author": {
		"name": "tldraw GB Ltd.",
		"email": "hello@tldraw.com"
	},
	"homepage": "https://tldraw.dev",
	"license": "Apache-2.0",
	"repository": {
		"type": "git",
		"url": "https://github.com/tldraw/tldraw"
	},
	"bugs": {
		"url": "https://github.com/tldraw/tldraw/issues"
	},
	"keywords": [
		"tldraw",
		"drawing",
		"app",
		"development",
		"whiteboard",
		"canvas",
		"infinite"
	],
	"/* NOTE */": "These `main` and `types` fields are rewritten by the build script. They are not the actual values we publish",
	"main": "./src/index.ts",
	"types": "./.tsbuild/index.d.ts",
	"style": "./editor.css",
	"/* GOTCHA */": "files will include ./dist and index.d.ts by default, add any others you want to include in here",
	"files": [
		"editor.css"
	],
	"scripts": {
		"test": "lazy inherit",
		"benchmark": "yarn run -T tsx ./scripts/benchmark.ts",
		"test-coverage": "lazy inherit",
		"build": "yarn run -T tsx ../../scripts/build-package.ts",
		"build-api": "yarn run -T tsx ../../scripts/build-api.ts",
		"prepack": "yarn run -T tsx ../../scripts/prepack.ts",
		"postpack": "../../scripts/postpack.sh",
		"pack-tarball": "yarn pack",
		"lint": "yarn run -T tsx ../../scripts/lint.ts"
	},
	"dependencies": {
		"@tldraw/state": "workspace:*",
		"@tldraw/store": "workspace:*",
		"@tldraw/tlschema": "workspace:*",
		"@tldraw/utils": "workspace:*",
		"@tldraw/validate": "workspace:*",
		"@types/core-js": "^2.5.5",
		"@use-gesture/react": "^10.2.27",
		"classnames": "^2.3.2",
		"core-js": "^3.31.1",
		"eventemitter3": "^4.0.7",
		"idb": "^7.1.1",
		"is-plain-object": "^5.0.0",
		"lodash.throttle": "^4.1.1",
		"lodash.uniq": "^4.5.0",
		"nanoid": "4.0.2"
	},
	"peerDependencies": {
		"react": "^18",
		"react-dom": "^18"
	},
	"devDependencies": {
		"@peculiar/webcrypto": "^1.4.0",
		"@testing-library/jest-dom": "^5.16.5",
		"@testing-library/react": "^14.0.0",
		"@types/benchmark": "^2.1.2",
		"@types/lodash.throttle": "^4.1.7",
		"@types/lodash.uniq": "^4.5.7",
		"@types/react-test-renderer": "^18.0.0",
		"@types/wicg-file-system-access": "^2020.9.5",
		"benchmark": "^2.1.4",
		"fake-indexeddb": "^4.0.0",
		"jest-canvas-mock": "^2.5.1",
		"jest-environment-jsdom": "^29.4.3",
		"lazyrepo": "0.0.0-alpha.27",
		"react-test-renderer": "^18.2.0",
		"resize-observer-polyfill": "^1.5.1"
	},
	"jest": {
		"preset": "config/jest/node",
		"testEnvironment": "jsdom",
		"fakeTimers": {
			"enableGlobally": true
		},
		"testPathIgnorePatterns": [
			"^.+\\.*.css$"
		],
		"transformIgnorePatterns": [
			"node_modules/(?!(nanoid)/)"
		],
		"moduleNameMapper": {
			"^~(.*)": "<rootDir>/src/$1",
			"\\.(css|less|scss|sass)$": "identity-obj-proxy"
		},
		"setupFiles": [
			"raf/polyfill",
			"jest-canvas-mock",
			"<rootDir>/setupTests.js"
		],
		"setupFilesAfterEnv": [
			"../../config/setupJest.ts"
		]
	}
}
