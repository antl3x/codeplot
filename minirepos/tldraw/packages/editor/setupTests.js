require('fake-indexeddb/auto')
global.ResizeObserver = require('resize-observer-polyfill')
global.crypto ??= new (require('@peculiar/webcrypto').Crypto)()
global.FontFace = class FontFace {
	load() {
		return Promise.resolve()
	}
}

document.fonts = {
	add: () => {},
	delete: () => {},
	forEach: () => {},
	[Symbol.iterator]: () => [][Symbol.iterator](),
}
