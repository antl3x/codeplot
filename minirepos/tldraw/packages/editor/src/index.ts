// Important! don't move this tlschema re-export to lib/index.ts, doing so causes esbuild to produce
// incorrect output. https://github.com/evanw/esbuild/issues/1737

export {
	EMPTY_ARRAY,
	atom,
	computed,
	react,
	track,
	transact,
	transaction,
	useComputed,
	useQuickReactor,
	useReactor,
	useValue,
	whyAmIRunning,
	type Atom,
	type Signal,
} from '@tldraw/state'
// eslint-disable-next-line local/no-export-star
export * from '@tldraw/store'
// eslint-disable-next-line local/no-export-star
export * from '@tldraw/tlschema'
// eslint-disable-next-line local/no-export-star
export * from '@tldraw/utils'
// eslint-disable-next-line local/no-export-star
export * from '@tldraw/validate'
export {
	ErrorScreen,
	LoadingScreen,
	TldrawEditor,
	type TLOnMountHandler,
	type TldrawEditorBaseProps,
	type TldrawEditorProps,
} from './lib/TldrawEditor'
export { Canvas } from './lib/components/Canvas'
export {
	ErrorBoundary,
	OptionalErrorBoundary,
	type TLErrorBoundaryProps,
} from './lib/components/ErrorBoundary'
export { HTMLContainer, type HTMLContainerProps } from './lib/components/HTMLContainer'
export { PositionedOnCanvas } from './lib/components/PositionedOnCanvas'
export { SVGContainer, type SVGContainerProps } from './lib/components/SVGContainer'
export { ShapeIndicator, type TLShapeIndicatorComponent } from './lib/components/ShapeIndicator'
export {
	DefaultBackground,
	type TLBackgroundComponent,
} from './lib/components/default-components/DefaultBackground'
export {
	DefaultBrush,
	type TLBrushComponent,
} from './lib/components/default-components/DefaultBrush'
export {
	DefaultCollaboratorHint,
	type TLCollaboratorHintComponent,
} from './lib/components/default-components/DefaultCollaboratorHint'
export {
	DefaultCursor,
	type TLCursorComponent,
} from './lib/components/default-components/DefaultCursor'
export { DefaultErrorFallback } from './lib/components/default-components/DefaultErrorFallback'
export { DefaultGrid, type TLGridComponent } from './lib/components/default-components/DefaultGrid'
export {
	DefaultHandle,
	type TLHandleComponent,
} from './lib/components/default-components/DefaultHandle'
export {
	DefaultHandles,
	type TLHandlesComponent,
} from './lib/components/default-components/DefaultHandles'
export {
	DefaultHoveredShapeIndicator,
	type TLHoveredShapeIndicatorComponent,
} from './lib/components/default-components/DefaultHoveredShapeIndicator'
export {
	DefaultScribble,
	type TLScribbleComponent,
} from './lib/components/default-components/DefaultScribble'
export {
	DefaultSelectionBackground,
	type TLSelectionBackgroundComponent,
} from './lib/components/default-components/DefaultSelectionBackground'
export {
	DefaultSelectionForeground,
	type TLSelectionForegroundComponent,
} from './lib/components/default-components/DefaultSelectionForeground'
export {
	DefaultSnapLine,
	type TLSnapLineComponent,
} from './lib/components/default-components/DefaultSnapLine'
export {
	DefaultSpinner,
	type TLSpinnerComponent,
} from './lib/components/default-components/DefaultSpinner'
export {
	DefaultSvgDefs,
	type TLSvgDefsComponent,
} from './lib/components/default-components/DefaultSvgDefs'
export {
	TAB_ID,
	createSessionStateSnapshotSignal,
	extractSessionStateFromLegacySnapshot,
	loadSessionStateSnapshotIntoStore,
	type TLSessionStateSnapshot,
} from './lib/config/TLSessionStateSnapshot'
export {
	USER_COLORS,
	defaultUserPreferences,
	getFreshUserPreferences,
	getUserPreferences,
	setUserPreferences,
	type TLUserPreferences,
} from './lib/config/TLUserPreferences'
export {
	createTLStore,
	type TLStoreEventInfo,
	type TLStoreOptions,
} from './lib/config/createTLStore'
export { createTLUser } from './lib/config/createTLUser'
export { coreShapes, type TLAnyShapeUtilConstructor } from './lib/config/defaultShapes'
export {
	ANIMATION_MEDIUM_MS,
	ANIMATION_SHORT_MS,
	CAMERA_SLIDE_FRICTION,
	DEFAULT_ANIMATION_OPTIONS,
	DOUBLE_CLICK_DURATION,
	DRAG_DISTANCE,
	GRID_STEPS,
	HASH_PATTERN_ZOOM_NAMES,
	HIT_TEST_MARGIN,
	MAX_PAGES,
	MAX_SHAPES_PER_PAGE,
	MAX_ZOOM,
	MIN_ZOOM,
	MULTI_CLICK_DURATION,
	SVG_PADDING,
	ZOOMS,
} from './lib/constants'
export {
	Editor,
	type TLAnimationOptions,
	type TLEditorOptions,
	type TLResizeShapeOptions,
} from './lib/editor/Editor'
export {
	SnapManager,
	type GapsSnapLine,
	type PointsSnapLine,
	type SnapLine,
	type SnapPoint,
} from './lib/editor/managers/SnapManager'
export { BaseBoxShapeUtil, type TLBaseBoxShape } from './lib/editor/shapes/BaseBoxShapeUtil'
export {
	ShapeUtil,
	type TLOnBeforeCreateHandler,
	type TLOnBeforeUpdateHandler,
	type TLOnBindingChangeHandler,
	type TLOnChildrenChangeHandler,
	type TLOnClickHandler,
	type TLOnDoubleClickHandleHandler,
	type TLOnDoubleClickHandler,
	type TLOnDragHandler,
	type TLOnEditEndHandler,
	type TLOnHandleChangeHandler,
	type TLOnResizeEndHandler,
	type TLOnResizeHandler,
	type TLOnResizeStartHandler,
	type TLOnRotateEndHandler,
	type TLOnRotateHandler,
	type TLOnRotateStartHandler,
	type TLOnTranslateEndHandler,
	type TLOnTranslateHandler,
	type TLOnTranslateStartHandler,
	type TLResizeInfo,
	type TLResizeMode,
	type TLShapeUtilCanvasSvgDef,
	type TLShapeUtilConstructor,
	type TLShapeUtilFlag,
} from './lib/editor/shapes/ShapeUtil'
export { GroupShapeUtil } from './lib/editor/shapes/group/GroupShapeUtil'
export { getArrowheadPathForType } from './lib/editor/shapes/shared/arrow/arrowheads'
export {
	getCurvedArrowHandlePath,
	getSolidCurvedArrowPath,
} from './lib/editor/shapes/shared/arrow/curved-arrow'
export { getArrowTerminalsInArrowSpace } from './lib/editor/shapes/shared/arrow/shared'
export {
	getSolidStraightArrowPath,
	getStraightArrowHandlePath,
} from './lib/editor/shapes/shared/arrow/straight-arrow'
export { resizeBox, type ResizeBoxOptions } from './lib/editor/shapes/shared/resizeBox'
export { BaseBoxShapeTool } from './lib/editor/tools/BaseBoxShapeTool/BaseBoxShapeTool'
export { StateNode, type TLStateNodeConstructor } from './lib/editor/tools/StateNode'
export { type SvgExportContext, type SvgExportDef } from './lib/editor/types/SvgExportContext'
export { type TLContent } from './lib/editor/types/clipboard-types'
export { type TLEventMap, type TLEventMapHandler } from './lib/editor/types/emit-types'
export {
	EVENT_NAME_MAP,
	type TLBaseEventInfo,
	type TLCLickEventName,
	type TLCancelEvent,
	type TLCancelEventInfo,
	type TLClickEvent,
	type TLClickEventInfo,
	type TLCompleteEvent,
	type TLCompleteEventInfo,
	type TLEnterEventHandler,
	type TLEventHandlers,
	type TLEventInfo,
	type TLEventName,
	type TLExitEventHandler,
	type TLInterruptEvent,
	type TLInterruptEventInfo,
	type TLKeyboardEvent,
	type TLKeyboardEventInfo,
	type TLKeyboardEventName,
	type TLPinchEvent,
	type TLPinchEventInfo,
	type TLPinchEventName,
	type TLPointerEvent,
	type TLPointerEventInfo,
	type TLPointerEventName,
	type TLPointerEventTarget,
	type TLTickEvent,
	type TLWheelEvent,
	type TLWheelEventInfo,
	type UiEvent,
	type UiEventType,
} from './lib/editor/types/event-types'
export {
	type TLExternalAssetContent,
	type TLExternalContent,
	type TLExternalContentSource,
} from './lib/editor/types/external-content'
export {
	type TLCommand,
	type TLCommandHandler,
	type TLHistoryEntry,
	type TLHistoryMark,
} from './lib/editor/types/history-types'
export { type RequiredKeys } from './lib/editor/types/misc-types'
export { type TLResizeHandle, type TLSelectionHandle } from './lib/editor/types/selection-types'
export { useContainer } from './lib/hooks/useContainer'
export { getCursor } from './lib/hooks/useCursor'
export { useEditor } from './lib/hooks/useEditor'
export type { TLEditorComponents } from './lib/hooks/useEditorComponents'
export { useIsCropping } from './lib/hooks/useIsCropping'
export { useIsDarkMode } from './lib/hooks/useIsDarkMode'
export { useIsEditing } from './lib/hooks/useIsEditing'
export { useLocalStore } from './lib/hooks/useLocalStore'
export { usePeerIds } from './lib/hooks/usePeerIds'
export { usePresence } from './lib/hooks/usePresence'
export { useSelectionEvents } from './lib/hooks/useSelectionEvents'
export { useTLStore } from './lib/hooks/useTLStore'
export { useTransform } from './lib/hooks/useTransform'
export {
	Box2d,
	ROTATE_CORNER_TO_SELECTION_CORNER,
	rotateSelectionHandle,
	type RotateCorner,
	type SelectionCorner,
	type SelectionEdge,
	type SelectionHandle,
} from './lib/primitives/Box2d'
export { Matrix2d, type Matrix2dModel } from './lib/primitives/Matrix2d'
export { Vec2d, type VecLike } from './lib/primitives/Vec2d'
export { EASINGS } from './lib/primitives/easings'
export { Arc2d } from './lib/primitives/geometry/Arc2d'
export { Circle2d } from './lib/primitives/geometry/Circle2d'
export { CubicBezier2d } from './lib/primitives/geometry/CubicBezier2d'
export { CubicSpline2d } from './lib/primitives/geometry/CubicSpline2d'
export { Edge2d } from './lib/primitives/geometry/Edge2d'
export { Ellipse2d } from './lib/primitives/geometry/Ellipse2d'
export { Geometry2d } from './lib/primitives/geometry/Geometry2d'
export { Group2d } from './lib/primitives/geometry/Group2d'
export { Polygon2d } from './lib/primitives/geometry/Polygon2d'
export { Polyline2d } from './lib/primitives/geometry/Polyline2d'
export { Rectangle2d } from './lib/primitives/geometry/Rectangle2d'
export { Stadium2d } from './lib/primitives/geometry/Stadium2d'
export {
	intersectLineSegmentPolygon,
	intersectLineSegmentPolyline,
	intersectPolygonPolygon,
	linesIntersect,
	polygonsIntersect,
} from './lib/primitives/intersect'
export {
	EPSILON,
	PI,
	PI2,
	SIN,
	TAU,
	angleDelta,
	approximately,
	areAnglesCompatible,
	average,
	canonicalizeRotation,
	clamp,
	clampRadians,
	clockwiseAngleDist,
	degreesToRadians,
	getArcLength,
	getPointOnCircle,
	getPolygonVertices,
	getStarBounds,
	getSweep,
	isAngleBetween,
	isSafeFloat,
	lerpAngles,
	longAngleDist,
	perimeterOfEllipse,
	pointInBounds,
	pointInCircle,
	pointInEllipse,
	pointInPolygon,
	pointInPolyline,
	pointInRect,
	pointNearToLineSegment,
	pointNearToPolyline,
	precise,
	radiansToDegrees,
	rangeIntersection,
	shortAngleDist,
	snapAngle,
	toDomPrecision,
	toFixed,
	toPrecision,
} from './lib/primitives/utils'
export {
	ReadonlySharedStyleMap,
	SharedStyleMap,
	type SharedStyle,
} from './lib/utils/SharedStylesMap'
export { WeakMapCache } from './lib/utils/WeakMapCache'
export { dataUrlToFile } from './lib/utils/assets'
export { debugFlags, featureFlags, type DebugFlag } from './lib/utils/debug-flags'
export {
	loopToHtmlElement,
	preventDefault,
	releasePointerCapture,
	setPointerCapture,
	stopEventPropagation,
} from './lib/utils/dom'
export { getIncrementedName } from './lib/utils/getIncrementedName'
export { getPointerInfo } from './lib/utils/getPointerInfo'
export { getSvgPathFromPoints } from './lib/utils/getSvgPathFromPoints'
export { hardResetEditor } from './lib/utils/hardResetEditor'
export { normalizeWheel } from './lib/utils/normalizeWheel'
export { refreshPage } from './lib/utils/refreshPage'
export {
	getIndexAbove,
	getIndexBelow,
	getIndexBetween,
	getIndices,
	getIndicesAbove,
	getIndicesBelow,
	getIndicesBetween,
	sortByIndex,
} from './lib/utils/reordering/reordering'
export {
	applyRotationToSnapshotShapes,
	getRotationSnapshot,
	type TLRotationSnapshot,
} from './lib/utils/rotation'
export { runtime, setRuntimeOverrides } from './lib/utils/runtime'
export { type TLStoreWithStatus } from './lib/utils/sync/StoreWithStatus'
export { hardReset } from './lib/utils/sync/hardReset'
export { uniq } from './lib/utils/uniq'
export { uniqueId } from './lib/utils/uniqueId'
export { openWindow } from './lib/utils/window-open'

/** @polyfills */

import 'core-js/stable/array/at'
import 'core-js/stable/array/flat'
import 'core-js/stable/array/flat-map'
import 'core-js/stable/string/at'
import 'core-js/stable/string/replace-all'
