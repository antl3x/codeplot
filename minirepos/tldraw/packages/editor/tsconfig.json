{
	"extends": "../../config/tsconfig.base.json",
	"include": ["src"],
	"exclude": ["node_modules", "dist", "**/*.css", ".tsbuild*"],
	"compilerOptions": {
		"outDir": "./.tsbuild",
		"rootDir": "src",
		// TODO: enable these two options
		"noImplicitOverride": false,
		"noImplicitReturns": false,
		"types": ["node", "@types/jest"]
	},
	"references": [
		{ "path": "../tlschema" },
		{ "path": "../store" },
		{ "path": "../validate" },
		{ "path": "../utils" },
		{ "path": "../state" }
	]
}
