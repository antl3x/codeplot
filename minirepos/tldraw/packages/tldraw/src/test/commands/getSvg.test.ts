import { DefaultDashStyle, SVG_PADDING, createShapeId } from '@tldraw/editor'
import { TestEditor } from '../TestEditor'

let editor: TestEditor

const ids = {
	boxA: createShapeId('boxA'),
	boxB: createShapeId('boxB'),
	boxC: createShapeId('boxC'),
}

beforeEach(() => {
	editor = new TestEditor()
	editor.setStyleForNextShapes(DefaultDashStyle, 'solid')
	editor.setStyleForSelectedShapes(DefaultDashStyle, 'solid')
	editor.createShapes([
		{
			id: ids.boxA,
			type: 'geo',
			x: 0,
			y: 0,
			props: {
				w: 100,
				h: 100,
				text: 'Hello world',
			},
		},
		{
			id: ids.boxB,
			type: 'geo',
			x: 100,
			y: 100,
			props: {
				w: 50,
				h: 50,
			},
		},
		{
			id: ids.boxC,
			type: 'geo',
			x: 400,
			y: 400,
			props: {
				w: 100,
				h: 100,
				fill: 'pattern',
			},
		},
	])
	editor.selectAll()
})

it('gets an SVG', async () => {
	const svg = await editor.getSvg(editor.selectedShapeIds)

	expect(svg).toBeTruthy()
})

it('Does not get an SVG when no ids are provided', async () => {
	const svg = await editor.getSvg([])

	expect(svg).toBeFalsy()
})

it('Gets the bounding box at the correct size', async () => {
	const svg = await editor.getSvg(editor.selectedShapeIds)
	const bbox = editor.selectionRotatedPageBounds!
	const expanded = bbox.expandBy(SVG_PADDING) // adds 32px padding

	expect(svg!.getAttribute('width')).toMatch(expanded.width + '')
	expect(svg!.getAttribute('height')).toMatch(expanded.height + '')
})

it('Gets the bounding box at the correct size', async () => {
	const svg = (await editor.getSvg(editor.selectedShapeIds))!
	const bbox = editor.selectionRotatedPageBounds!
	const expanded = bbox.expandBy(SVG_PADDING) // adds 32px padding

	expect(svg!.getAttribute('width')).toMatch(expanded.width + '')
	expect(svg!.getAttribute('height')).toMatch(expanded.height + '')
})

it('Matches a snapshot', async () => {
	const svg = (await editor.getSvg(editor.selectedShapeIds))!

	const elm = document.createElement('wrapper')
	elm.appendChild(svg)

	expect(elm).toMatchSnapshot('Basic SVG')
})

it('Accepts a scale option', async () => {
	const svg1 = (await editor.getSvg(editor.selectedShapeIds, { scale: 1 }))!

	expect(svg1.getAttribute('width')).toBe('564')

	const svg2 = (await editor.getSvg(editor.selectedShapeIds, { scale: 2 }))!

	expect(svg2.getAttribute('width')).toBe('1128')
})

it('Accepts a background option', async () => {
	const svg1 = (await editor.getSvg(editor.selectedShapeIds, { background: true }))!

	expect(svg1.style.backgroundColor).not.toBe('transparent')

	const svg2 = (await editor.getSvg(editor.selectedShapeIds, { background: false }))!

	expect(svg2.style.backgroundColor).toBe('transparent')
})
