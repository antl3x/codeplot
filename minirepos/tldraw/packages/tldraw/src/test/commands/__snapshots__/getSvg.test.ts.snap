// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches a snapshot: Basic SVG 1`] = `
<wrapper>
  <svg
    direction="ltr"
    height="564"
    stroke-linecap="round"
    stroke-linejoin="round"
    style="background-color: transparent;"
    viewBox="-32 -32 564 564"
    width="564"
  >
    <defs>
      <!--def: tldraw:font:pattern-->
      <mask
        id="hash_pattern_mask"
      >
        
							
        <rect
          fill="white"
          height="8"
          width="8"
          x="0"
          y="0"
        />
        
							
        <g
          stroke="black"
          strokelinecap="round"
        >
          
								
          <line
            x1="0.6666666666666666"
            x2="2"
            y1="2"
            y2="0.6666666666666666"
          />
          
								
          <line
            x1="3.333333333333333"
            x2="4.666666666666666"
            y1="4.666666666666666"
            y2="3.333333333333333"
          />
          
								
          <line
            x1="6"
            x2="7.333333333333333"
            y1="7.333333333333333"
            y2="6"
          />
          
							
        </g>
        
						
      </mask>
      <pattern
        height="8"
        id="hash_pattern"
        patternUnits="userSpaceOnUse"
        width="8"
      >
        
							
        <rect
          fill="#fcfffe"
          height="8"
          mask="url(#hash_pattern_mask)"
          width="8"
          x="0"
          y="0"
        />
        
						
      </pattern>
    </defs>
    <g
      opacity="1"
      transform="matrix(1, 0, 0, 1, 0, 0)"
    >
      <g>
        <path
          d="M0, 0L100, 0,100, 428,0, 428Z"
          fill="none"
          stroke="#1d1d1d"
          stroke-width="3.5"
        />
        <g>
          <text
            alignment-baseline="mathematical"
            dominant-baseline="mathematical"
            fill="rgb(249, 250, 251)"
            font-family="'tldraw_draw', sans-serif"
            font-size="22px"
            font-style="normal"
            font-weight="normal"
            line-height="29.700000000000003px"
            stroke="rgb(249, 250, 251)"
            stroke-width="2"
          >
            <tspan
              alignment-baseline="mathematical"
              x="16px"
              y="-17px"
            >
              Hello world
            </tspan>
          </text>
          <text
            alignment-baseline="mathematical"
            dominant-baseline="mathematical"
            fill="#1d1d1d"
            font-family="'tldraw_draw', sans-serif"
            font-size="22px"
            font-style="normal"
            font-weight="normal"
            line-height="29.700000000000003px"
            stroke="none"
          >
            <tspan
              alignment-baseline="mathematical"
              x="16px"
              y="-17px"
            >
              Hello world
            </tspan>
          </text>
        </g>
      </g>
    </g>
    <g
      opacity="1"
      transform="matrix(1, 0, 0, 1, 100, 100)"
    >
      <path
        d="M0, 0L50, 0,50, 50,0, 50Z"
        fill="none"
        stroke="#1d1d1d"
        stroke-width="3.5"
      />
    </g>
    <g
      opacity="1"
      transform="matrix(1, 0, 0, 1, 400, 400)"
    >
      <g>
        <g>
          <path
            d="M0, 0L100, 0,100, 100,0, 100Z"
            fill="#494949"
          />
          <path
            d="M0, 0L100, 0,100, 100,0, 100Z"
            fill="url(#hash_pattern)"
          />
        </g>
        <path
          d="M0, 0L100, 0,100, 100,0, 100Z"
          fill="none"
          stroke="#1d1d1d"
          stroke-width="3.5"
        />
      </g>
    </g>
  </svg>
</wrapper>
`;
